app/core/model/src/main/java/io/syndesis/model/metrics/IntegrationMetricsSummary.java

```diff
@@ -53,6 +54,8 @@ default Kind getKind() {
      */
     Optional<Date> getLastProcessed();
 
+    Optional<List<IntegrationMetricsSummary>> getIntegrationDeploymentMetrics();
+
```

Comment:
well, not  really better, as this allows for arbitrary deep nesting, and a `getIntegrationDeploymentMetrics` on a deployment metrics object itself doesn't make sense. Also an IntegrationDeploymentMetrics is not a IntegrationMetricsSummary as modelled here.

If you insist of nesting, then you should use a subclass with `IntegrationMetricsSummary extends IntegrationMetrics` and adding a `List<IntegrationMetrics> getIntegrationDeploymentMetrics()` to this subclass.

Whats so bad about having a flat list with summary and list of detailede deployment metrics side-by-side ? (don't really understand also, why a summary should have a list of details)